File: Python/algorytmy/potega.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#
#  potega.py
#  Obliczanie potęgi liczby naturalnej

def potega_re(a, n):
    if n == 0:
        return 1
    potega_re(a, n)
        

def potega_it(a, n):
    wynik = 1
    
    for i in range(n):
        wynik = wynik * a
        #print (wynik)

    return wynik

def main(args):
    a, n = 3, 0 # wielokrotne przypisanie
    print("Podstawa {} do potęgi {} wynosi {}".
           format(a, n, potega_it(a, n))) # wywołanie funkcji

    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/potega.cpp
/*
 * potega.cpp
 */

#include <iostream>
using namespace std;


int potega_it(int a, int n) {
    int wynik = 1;
    for (int i = 0; i < n; i++){
        wynik = wynik * a;
    }
    return wynik;
}

int potega_re(int a, int n) {
    if (n==0)
        return 1;
    return potega_re(a, n-1) * a;
}

int main(int argc, char **argv)
{
	int a, n;
    cout << "Podaj podstawę i wykładnik: ";
    cin >> a >> n;
    cout << "Potęga: " << potega_it(a, n) << endl;
    cout << "Potęga: " << potega_re(a, n) << endl;
	return 0;
}

